include/hcc_debug_messages.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <iostream>
+#include <string>
+#include <stdio.h>
+
+#define DB_MISC         1U << 0
+#define DB_SYNC         1U << 1
+#define DB_AQL          1U << 2
+#define DB_QUEUE        1U << 3
+
+#define DEBUG_ENABLED 1
+
+char* hcc_db;
+std::ostream &err = std::cerr;
```

Comment:
considering the cast where multiple object files may include this header file, could you put them into an anonymous namespace to prevent from breaking ODR rule?